### PR TITLE
Mpd 1444 report submission scenarios

### PR DESCRIPTION
**Pull request by Winsto**
First part of rewrite of 'end to end' integration tests; these cover report submission. In terms of 'end points' these are the same as the previous tests; ReportController.Submit as the 'front end point'. We may need to cover some other behaviour in terms of re-directs for partially completed sections/questions.

### CODE REVIEW

File: src/SFA.DAS.PSRService.Web/Services/IReportService.cs [hunk 1]
```diff
@@ -2,15 +2,17 @@
...
-using SFA.DAS.PSRService.Web.Models;
-using SFA.DAS.PSRService.Web.ViewModels;
 
 namespace SFA.DAS.PSRService.Web.Services
 {
     public interface IReportService
     {
         void CreateReport(string employerId);
-        Report GetReport(string period,string employerId);
+
```
Comment (kuzinator) [inline src/SFA.DAS.PSRService.Web/Services/IReportService.cs]: Should this be rolled back or refactored?

Comment (Winsto) [reply src/SFA.DAS.PSRService.Web/Services/IReportService.cs]: A start was made on replacing the 'stringly typed' GetReport method with one using the, newly introduced, Period object. However those changes were backed out, so we are left with an unused method. The work should be done to replace the stringly typed method, but it makes sense to remove it for now.

Removed.

File: src/SFA.DAS.PSRService.IntegrationTests/ReportSubmission/Given_I_Have_Created_A_Report/And_I_Have_Reported_All_Mandatory_Factors/And_All_Mandatory_Numbers/When_I_Sumit_The_Report.cs [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+{
+    public sealed class When_I_Sumit_The_Report
+        : And_All_Mandatory_Numbers
+    {
+        private IActionResult submitResponse;
+
+        protected override void When()
+        {
+            submitResponse
+                =
```
Comment (desderry) [inline src/SFA.DAS.PSRService.IntegrationTests/ReportSubmission/Given_I_Have_Created_A_Report/And_I_Have_Reported_All_Mandatory_Factors/And_All_Mandatory_Numbers/When_I_Sumit_The_Report.cs]: can we tidy this up as its hard to read at the moment spread over multiple lines

File: src/SFA.DAS.PSRService.IntegrationTests/ReportSubmission/Given_I_Have_Created_A_Report/And_I_Have_Reported_All_Mandatory_Factors/And_All_Mandatory_Numbers/When_I_Sumit_The_Report.cs [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+                SUT
+                    .Submit();
+        }
+
+        [Test]
+        public void Then_I_Am_Presented_With_The_Submitted_View()
+        {
+            Assert
+                .AreEqual(
```
Comment (desderry) [inline src/SFA.DAS.PSRService.IntegrationTests/ReportSubmission/Given_I_Have_Created_A_Report/And_I_Have_Reported_All_Mandatory_Factors/And_All_Mandatory_Numbers/When_I_Sumit_The_Report.cs]: Would prefer this over 1 or 2 lines to ease readability

File: src/SFA.DAS.PSRService.IntegrationTests/ReportSubmission/Given_I_Have_Created_A_Report/And_I_Have_Reported_All_Mandatory_Factors/But_Only_Some_Mandatory_Numbers/When_I_Submit_The_Report.cs [hunk 4]
```diff
@@ -0,0 +1,59 @@
...
+{
+    public sealed class When_I_Submit_The_Report
+    :But_Only_Some_Mandatory_Numbers
+    {
+        private IActionResult submitResult;
+
+        protected override void When()
+        {
+            submitResult
+                =
```
Comment (desderry) [inline src/SFA.DAS.PSRService.IntegrationTests/ReportSubmission/Given_I_Have_Created_A_Report/And_I_Have_Reported_All_Mandatory_Factors/But_Only_Some_Mandatory_Numbers/When_I_Submit_The_Report.cs]: as above, tidy up so not over multiple lines

File: src/SFA.DAS.PSRService.IntegrationTests/ReportSubmission/Given_I_Have_Created_A_Report/And_I_Have_Reported_All_Mandatory_Factors/But_Only_Some_Mandatory_Numbers/When_I_Submit_The_Report.cs [hunk 5]
```diff
@@ -0,0 +1,59 @@
...
+        [Test]
+        public void Then_I_Am_Redirected_To_The_Home_Index()
+        {
+            Assert
+                .IsInstanceOf<RedirectResult>(
+                    submitResult);
+
+            MockUrlHelper
+                .Verify(
+                    m =>
```
Comment (desderry) [inline src/SFA.DAS.PSRService.IntegrationTests/ReportSubmission/Given_I_Have_Created_A_Report/And_I_Have_Reported_All_Mandatory_Factors/But_Only_Some_Mandatory_Numbers/When_I_Submit_The_Report.cs]: would combining this with the line above and below help out with readability?

File: src/SFA.DAS.PSRService.IntegrationTests/ReportSubmission/Given_I_Have_Created_A_Report/And_I_Have_Reported_All_Mandatory_Numbers/But_Only_One_Mandatory_Factor/When_I_Submit_The_Report.cs [hunk 6]
```diff
@@ -0,0 +1,59 @@
...
+{
+    public sealed class When_I_Submit_The_Report
+    : But_Only_One_Mandatory_Factor
+    {
+        private IActionResult submitResult;
+
+        protected override void When()
+        {
+            submitResult
+                =
```
Comment (desderry) [inline src/SFA.DAS.PSRService.IntegrationTests/ReportSubmission/Given_I_Have_Created_A_Report/And_I_Have_Reported_All_Mandatory_Numbers/But_Only_One_Mandatory_Factor/When_I_Submit_The_Report.cs]: change to single line

File: src/SFA.DAS.PSRService.IntegrationTests/ReportSubmission/Given_I_Have_Created_A_Report/And_I_Have_Reported_Some_Mandatory_Numbers/And_Some_Mandatory_Factors/When_I_Submit_The_Report.cs [hunk 7]
```diff
@@ -0,0 +1,59 @@
...
+{
+    public sealed class When_I_Submit_The_Report
+    : And_Some_Mandatory_Factors
+    {
+        private IActionResult submitResult;
+
+        protected override void When()
+        {
+            submitResult
+                =
```
Comment (desderry) [inline src/SFA.DAS.PSRService.IntegrationTests/ReportSubmission/Given_I_Have_Created_A_Report/And_I_Have_Reported_Some_Mandatory_Numbers/And_Some_Mandatory_Factors/When_I_Submit_The_Report.cs]: change to one line

**Comment by Winsto** [comment]
@desderry I will comment here, rather than address each comment individually. New commit should have addressed all vertical vs horizontal layout points you raised.

**Pull request merged**